### PR TITLE
Ancient8 deploy

### PR DESCRIPTION
**Pull request by tkporter**
### Description

- Deploys ancient8 core
- Deploys a USDC warp route from Ethereum to ancient8
- Some changes to the key generation step - specifically:
  - `scripts/create-keys.ts` now expects `--newChainValidators chainname=1` instead of the flag `--newChains`
  - fixed the logic around adding new validators, which previously was broken
- Adds a hook module to deploy.ts - this is meant to be used in a similar fashion to the warp module, where configs are built ad-hoc
- changes `aw-multisig.json` to be `aw-multisig-${context}.json`, as previously the entire aw-multisig.json would be overridden no matter what the current context was

### CODE REVIEW

File: typescript/infra/config/environments/mainnet3/core/verification.json [hunk 1]
```diff
@@ -1,2144 +1,2830 @@
 {
```
Comment (tkporter) [inline typescript/infra/config/environments/mainnet3/core/verification.json:1]: the reason for this big diff is we moved to deterministic JSON stringifying in a previous PR, and this hadn't been applied yet to verification.jsons. I used https://jsonformatter.org/json-compare to make sure that the only value diff here is just the addition of `ancient8`

File: typescript/infra/config/environments/mainnet3/ism/verification.json [hunk 2]
```diff
@@ -1,2310 +1,2372 @@
 {
```
Comment (tkporter) [inline typescript/infra/config/environments/mainnet3/ism/verification.json:1]: same as core verification:

the reason for this big diff is we moved to deterministic JSON stringifying in a previous PR, and this hadn't been applied yet to verification.jsons. I used https://jsonformatter.org/json-compare to make sure that the only value diff here is just the addition of `ancient8`

File: rust/config/mainnet3_config.json [hunk 3]
```diff
@@ -1,6 +1,61 @@
 {
   "chains": {
+    "ancient8": {
+      "aggregationHookFactory": "0xEb9FcFDC9EfDC17c1EC5E1dc085B98485da213D6",
```
Comment (paulbalaji) [inline rust/config/mainnet3_config.json:4]: pls do the renamings as done in https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3471

File: typescript/infra/config/aw-multisig-rc.json
Comment (yorhodes) [inline typescript/infra/config/aw-multisig-rc.json:1]: surprised we have 3 validators on all these for rc but nbd

File: typescript/infra/scripts/print-token-prices.ts [hunk 4]
```diff
@@ -20,7 +20,19 @@ async function main() {
 
   const idPrices = await resp.json();
 
-  const prices = objMap(ids, (_, id) => idPrices[id][CURRENCY].toString());
+  const prices = objMap(ids, (_, id) => {
+    const idData = idPrices[id];
+    if (!idData) {
+      throw new Error(
+        `No data for ${id}, did you set gasCurrencyCoinGeckoId in the metadata?`,
+      );
```
Comment (yorhodes) [inline typescript/infra/scripts/print-token-prices.ts:28]: nice

File: typescript/infra/scripts/print-token-prices.ts [hunk 5]
```diff
@@ -20,7 +20,19 @@ async function main() {
...
+    }
+    const price = idData[CURRENCY];
+    if (!price) {
+      throw new Error(`No price for ${id}`);
```
Comment (yorhodes) [inline typescript/infra/scripts/print-token-prices.ts:32]: nit
```suggestion
      throw new Error(`No ${CURRENCY} price for ${id}`);
```

File: typescript/infra/scripts/deploy.ts [hunk 6]
```diff
@@ -121,19 +128,46 @@ async function main() {
       multiProvider,
     );
     const routerConfig = core.getRouterConfig(envConfig.owners);
-    const inevm = {
-      ...routerConfig.inevm,
-      type: TokenType.native,
-      interchainSecurityModule: ethers.constants.AddressZero,
-      owner: safes.inevm,
+
+    const buildIsmConfig = (remote: string): IsmConfig => ({
```
Comment (yorhodes) [inline typescript/infra/scripts/deploy.ts:132]: I think we already have a function like this

File: typescript/infra/scripts/deploy.ts [hunk 7]
```diff
@@ -121,19 +128,46 @@ async function main() {
...
+          ...defaultMultisigConfigs[remote],
+        },
+      ],
+    });
+
+    const ethereum = {
+      ...routerConfig.ethereum,
+      type: TokenType.collateral,
+      // USDC
+      token: '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48',
```
Comment (yorhodes) [inline typescript/infra/scripts/deploy.ts:151]: honestly we should have collateral consts for this stuff

File: typescript/infra/scripts/deploy.ts [hunk 8]
```diff
@@ -194,6 +228,24 @@ async function main() {
...
+      ismFactory,
+    );
+    // Config is intended to be changed for ad-hoc use cases:
+    config = {
+      ethereum: {
+        ...(coreConfig.ethereum.defaultHook as FallbackRoutingHookConfig)
+          .domains.ancient8,
+        owner: DEPLOYER,
+      },
+    };
```
Comment (yorhodes) [inline typescript/infra/scripts/deploy.ts:248]: interesting, I kind of expected this to be `core.defaultHook` but thats also a bit awkward

**Review by yorhodes — APPROVED** [review]

**Review by jmrossy — APPROVED** [review]
SDK changes lgtm

File: typescript/infra/config/environments/mainnet3/warp/addresses.json
Comment (paulbalaji) [inline typescript/infra/config/environments/mainnet3/warp/addresses.json:1]: is this information in registry?

File: typescript/infra/config/aw-multisig-rc.json
Comment (tkporter) [reply typescript/infra/config/aw-multisig-rc.json:1]: yeah probably worth cutting these down at some point

File: typescript/infra/scripts/deploy.ts (hunk 8 above)
Comment (tkporter) [reply typescript/infra/scripts/deploy.ts:248]: honestly I don't totally remember why we did it like this but I feel like we had a good reason, something to do with contract ownership or keeping the deployed hook as simple as possible

File: typescript/infra/config/environments/mainnet3/warp/addresses.json
Comment (tkporter) [reply typescript/infra/config/environments/mainnet3/warp/addresses.json:1]: no 😛

**Pull request merged**